janus.c

```diff
@@ -3395,8 +3415,40 @@ gboolean carbyne_janus_transport_is_sanityhealthcheck_resources_available(janus_
         }
     }
 	fclose(videoRoomStatusFile);
+
+	//checking RAM usage
+	FILE *meminfo = NULL;
+    char *line = NULL;
+    size_t len = 0;
+    ssize_t read = 0 ;
+    long mem[3];
+    meminfo = fopen("/proc/meminfo", "r");
+    if (meminfo == NULL)
+    {
+        JANUS_LOG(LOG_ERR,"Failed opening file\n");
+        return 0;
+    }
+    int index = 0;
+    while ((read = getline(&line, &len, meminfo)) != -1)
+    {
+        if (strstr(line, "Mem") != NULL && index < 3)
+        {
+            mem[index] = getLong(line);
+            index++;
+        }
+    }
+    fclose(meminfo);
+
+    int percentageFree = mem[2] * 100 / mem[0];
```

Comment:
Before using the return values from `/proc/meminfo` please verify them explicitly:
Check they contain what you expect them to contain:
```
MemTotal
MemAvailable
```

Reply:
What do you mean, the file is a constant structure in unix system it will always be the same other than the values.
here is how it should look
# cat /proc/meminfo 
MemTotal:        1882064 kB
MemFree:         1376380 kB
MemAvailable:    1535676 kB
Buffers:            2088 kB
Cached:           292324 kB
SwapCached:            0 kB
Active:           152944 kB
Inactive:         252628 kB
Active(anon):     111328 kB
Inactive(anon):    16508 kB
Active(file):      41616 kB
Inactive(file):   236120 kB
Unevictable:           0 kB
Mlocked:               0 kB
SwapTotal:       2097148 kB
SwapFree:        2097148 kB
Dirty:                40 kB
Writeback:             0 kB
AnonPages:        111180 kB
Mapped:            56396 kB
Shmem:             16676 kB
Slab:              54508 kB
SReclaimable:      25456 kB
SUnreclaim:        29052 kB
KernelStack:        2608 kB
PageTables:         5056 kB
NFS_Unstable:          0 kB
Bounce:                0 kB
WritebackTmp:          0 kB
CommitLimit:     3038180 kB
Committed_AS:     577664 kB
VmallocTotal:   34359738367 kB
VmallocUsed:       14664 kB
VmallocChunk:   34359717628 kB
HardwareCorrupted:     0 kB
AnonHugePages:     24576 kB
HugePages_Total:       0
HugePages_Free:        0
HugePages_Rsvd:        0
HugePages_Surp:        0
Hugepagesize:       2048 kB
DirectMap4k:       69632 kB
DirectMap2M:     2027520 kB

Reply:
I know.
Still, the best practice is to verify that you got what you expected to get.
Things can change in new versions, could be errors in reading, etc.
This is the best practice. We should verify what we get.